lib/imageflow/graph_runner.ex

```diff
@@ -1,14 +1,36 @@
 defmodule Imageflow.GraphRunner do
-  alias Imageflow.{Graph, Native}
+  alias Imageflow.{Graph, Native, Result}
 
   def run(%Graph{} = graph) do
     with {:ok, job} <- Native.create(),
          :ok <- add_inputs(job, graph.inputs),
          :ok <- add_outputs(job, graph.outputs),
          :ok <- send_task(job, graph),
-         :ok <- save_outputs(job, graph.outputs),
-         :ok <- Native.destroy(job) do
-      :ok
+         :ok <- save_outputs(job, graph.outputs) do
+      {:ok, job, graph}
+    end
+  end
+
+  def get_results(job, %Graph{outputs: outputs} = graph) do
+    outputs
+    |> Enum.reduce_while({:ok, []}, fn {id, value}, {:ok, _acc} ->
+      case value do
+        :bytes ->
+          case Native.get_output_buffer(job, id) do
+            {:ok, results} -> {:cont, {:ok, :binary.list_to_bin(results)}}
+            {:error, _} = error -> {:halt, error}
+          end
+
+        {:file, path} ->
+          {:cont, {:ok, path}}
+      end
```

Comment:
I might be missreading this, but isnt' this resulting in an `:output` that corresponds just to whatever the last output from the job was?
I'm guessing it should instead build a map of all the existing outputs